### PR TITLE
Add ability to select objects when a scene is loaded

### PR DESCRIPTION
**Pull request by sercero**

### CODE REVIEW

**Comment by paroj** [comment]
first of all, I really like the approach you have taken to integrate object selection.

That said, I have some nitpicking again.. xD

File: ogre_mesh_viewer.py [hunk 1]
```diff
@@ -253,7 +252,8 @@ def preRenderTargetUpdate(self, evt):
 
         self.logwin.draw()
 
-        if self.app.attach_node is not None:
+        #if self.app.attach_node is not None:
+        if entity == None:
```
Comment (paroj) [inline ogre_mesh_viewer.py:256]: https://pylint.readthedocs.io/en/latest/user_guide/messages/convention/singleton-comparison.html

also do not leave commented code behind

File: ogre_mesh_viewer.py [hunk 2]
```diff
@@ -313,6 +313,63 @@ def preRenderTargetUpdate(self, evt):
             entity.getSubEntities()[highlight].setMaterial(self.app.highlight_mat)
             self.highlighted = highlight
 
+        if ImGui.CollapsingHeader("Transform"):
```
Comment (paroj) [inline ogre_mesh_viewer.py:316]: I think this should go at the very end after "Bounds".

Also do not display this when only a mesh is loaded. In this case it is always attached to the root node and the transform is identity. So this tab does nothing.

File: ogre_mesh_viewer.py [hunk 3]
```diff
@@ -313,6 +313,63 @@ def preRenderTargetUpdate(self, evt):
...
+            flags = ImGui.TableFlags_Borders | ImGui.TableFlags_SizingStretchProp
+
+            enode = entity.getParentSceneNode()
+
+            ImGui.Text("Position")
+            if ImGui.BeginTable("Position", 3, flags):
```
Comment (paroj) [inline ogre_mesh_viewer.py:322]: I think using Tables here is too bloated both in terms of code and in terms of UI space.

Can you change this to BulletText as in "Bounds"?

File: ogre_mesh_viewer.py [hunk 4]
```diff
@@ -313,6 +313,63 @@ def preRenderTargetUpdate(self, evt):
...
+                ImGui.Text(str(round(enode.getPosition().y, 2)))
+                ImGui.TableNextColumn()
+                ImGui.Text(str(round(enode.getPosition().z, 2)))
+                ImGui.EndTable()
+
+            ImGui.Separator()
+
+            ImGui.Text("Orientation")
+            if ImGui.BeginTable("Orientation", 4, flags):
+                ImGui.TableSetupColumn("w")
```
Comment (paroj) [inline ogre_mesh_viewer.py:340]: I dont think that seeing the raw quaternion values is very helpful.
You should rather print pitch, yaw, roll as those are the rotations around local x,y,z axes.

File: ogre_mesh_viewer.py [hunk 5]
```diff
@@ -339,7 +396,7 @@ def preRenderTargetUpdate(self, evt):
                     changed = False
                     if astate.getLength() > 0:
                         ImGui.SameLine()
-                        changed, value = ImGui.SliderFloat("", astate.getTimePosition(), 0, astate.getLength(), "%.3fs")
+                        changed, value = ImGui.SliderFloat("", astate.getTimePosition(), 0, astate.getLength(), "%.2fs")
```
Comment (paroj) [inline ogre_mesh_viewer.py:399]: unrelated change? Please keep .3f

File: ogre_mesh_viewer.py [hunk 6]
```diff
@@ -434,20 +489,29 @@ def keyPressed(self, evt):
     def mousePressed(self, evt):
         if evt.clicks != 2:
```
Comment (paroj) [inline ogre_mesh_viewer.py:490]: I would allow selection with a single click, as double click is not intuitive

File: ogre_mesh_viewer.py [hunk 7]
```diff
@@ -434,20 +489,29 @@ def keyPressed(self, evt):
...
+                return True
+
+            if hit.movable and evt.button == OgreBites.BUTTON_LEFT:
+                if hit.movable.getMovableType() == "Entity":
+                    if self.entity != None:
+                        self.entity.getParentSceneNode().showBoundingBox(False)
+
+                    self.entity = self.scn_mgr.getEntity(hit.movable.getName())
+                    self.entity.getParentSceneNode().showBoundingBox(True)
+                    self.camman.setPivotOffset(ray.getPoint(hit.distance))
```
Comment (paroj) [inline ogre_mesh_viewer.py:509]: the camera should not move if you switch selection with a single click

File: ogre_mesh_viewer.py (hunk 3 above)
Comment (sercero) [reply ogre_mesh_viewer.py:322]: Most game engines show this in tabbed format and even with different colors for each axis.

I think the Bulleted format looks bad

File: ogre_mesh_viewer.py (hunk 6 above)
Comment (sercero) [reply ogre_mesh_viewer.py:490]: I was even thinking of having a sort of highlight on hover using a red color, like the submeshes

File: ogre_mesh_viewer.py (hunk 7 above)
Comment (sercero) [reply ogre_mesh_viewer.py:509]: But it centers the camera on the selected object...

File: ogre_mesh_viewer.py (hunk 5 above)
Comment (sercero) [reply ogre_mesh_viewer.py:399]: True, but is it necessary to have so much precision.?

It looks bad and reformats the text too much when playing.

File: ogre_mesh_viewer.py (hunk 7 above)
Comment (paroj) [reply ogre_mesh_viewer.py:509]: imagine this happening in blender

Comment (sercero) [reply ogre_mesh_viewer.py:509]: True, but in Blender you're not necesarily interested in inspecting the meshes which is the purpose of the mesh-viewer... that is my logic at least.

Comment (paroj) [reply ogre_mesh_viewer.py:509]: yes, but when inspecting the scene, you do not necessary want to switch focus to a specific mesh if you e.g. just checking the transforms. For switching focus, there is the double click - which incidentally will also change selection

File: ogre_mesh_viewer.py (hunk 3 above)
Comment (paroj) [reply ogre_mesh_viewer.py:322]: please go with the bulleted format for now. I have an idea how to improve formatting closer to

unity:
grafik

and unreal:
grafik

I will apply that to transform and bounds in a second step
